File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        ans = 0 
        n = len(heights)   
        for i in range(n): 
            l = r = i
            while l > 0 and heights[l -1] >= heights[i]: 
                l -= 1 
 
            while r < n-1 and heights[r+1] >= heights[i]:
                r += 1

            print(f"{l} - {i} - {r}, {heights[i]} * {r -l +1}")
            ans = max(ans, heights[i] * (r -l +1))
        return ans 
                    

File: Data Structures & Algorithms/max-water-container/submission-0.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int ans = 0 ; 

        for(int i =0; i<heights.Length; i++){
            for(int j = i +1 ; j < heights.Length; j++){
                ans = Math.Max(ans, (j-i) * Math.Min(heights[i], heights[j]) );
            }
        }

        return ans ;

    }
}
